### PR TITLE
Users/nnmkhang/add search to frontend

### PR DESCRIPTION
**Pull request by nnmkhang**
Overview of changes in this PR:
- Added Filter / Sort dropdown
- Added Redux states for Filter/Sort
- Added new component called Search Page 

Notes: 
- We are still deciding on what design the mobile view will have so I have not accommodated for the mobile view.
- The Filter/Sort actions currently does not support api calls to the backend. Functionality for this will be coming in a PR soon.
- Filter/Sort currently does not clear when the page is refreshed/switching between pages. This will be fixed through the search state in a later PR coming soon. ( This will be the same PR that hides the search bar when switching to pages which do not require the search bar, ex: New Template, etc ) 
- Displaying Cards when searched is not implemented in this PR, this functionality will come in a PR coming soon. 

Images:

Picture of Filter/Sort default state with new SearchPage component:
image

Picture of sort dropdown working: 
image

Picture of filter dropdown working:
image

### CODE REVIEW

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -46,16 +48,25 @@ class Dashboard extends React.Component<Props, State> {
 
   toggleModal = () => {
     this.setState({ isPreviewOpen: !this.state.isPreviewOpen });
-  }
+  };
 
   render() {
+
+    if(this.props.isSearch === true) {
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: can remove the == true?

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -34,10 +36,10 @@ interface Props {
   user?: UserType;
   authButtonMethod: () => Promise<void>;
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: I dont think this is being populated or used

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -46,16 +48,25 @@ class Dashboard extends React.Component<Props, State> {
...
+        <DashboardContainer>
+          <SearchPage/>
+        </DashboardContainer>
+      );
+    }
     return (
       <DashboardContainer>
         <Title>Recent</Title>
         <Gallery toggleModal={this.toggleModal}></Gallery>
+        <PreviewModal show={this.state.isPreviewOpen}toggleModal={this.toggleModal}/>
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: nit: space

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -46,16 +48,25 @@ class Dashboard extends React.Component<Props, State> {
...
         <Title>Drafts</Title>
         <Gallery toggleModal={this.toggleModal}></Gallery>
-        <PreviewModal show={this.state.isPreviewOpen} toggleModal={this.toggleModal} />
+        <PreviewModal show={this.state.isPreviewOpen} toggleModal={this.toggleModal}/>
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: Why is there two preview modals?

File: private-templates-service/client/src/components/Dashboard/SearchPage/Filter/Filter.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+interface Props {
+  isSearch: boolean;
+  filterType: string; 
+  queryFilter: (filterType: string) => void;
+  clearFilter: () => void;
+}
+
+const options: IDropdownOption[] = [
+  {key: 'created by me', text: "Created by me"},
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/SearchPage/Filter/Filter.tsx]: Have we implemented some sort of localization yet?

File: private-templates-service/client/src/components/Dashboard/SearchPage/SearchPage.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+class SearchPage extends React.Component<Props> {
+  constructor(props: Props){
+    super(props);
+  };
+
+  render() {
+    return (
+      <div>
+      <SearchResultBanner>
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/SearchPage/SearchPage.tsx]: nit: indentation

File: private-templates-service/client/src/components/Dashboard/SearchPage/styled.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+export const SearchResultBanner = styled.div`
+  display:flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 20px 0;
+`;
+
+export const StyledSearchText = styled.div`
+  letter-spacing 0;
+  opacity: 1;
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/SearchPage/styled.tsx]: opacity default is 1, should be able to remove this

File: private-templates-service/client/src/components/Dashboard/SearchPage/styled.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  font-size: 26px;
+  margin-bottom: 0;
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/SearchPage/styled.tsx]: margin-bottom default is 0

File: private-templates-service/client/src/components/Dashboard/Dashboard.tsx (hunk 4 above)
Comment (nnmkhang) [reply private-templates-service/client/src/components/Dashboard/Dashboard.tsx]: image

there are two preview modals, one for each gallery ( Recent and Drafts ). I think we may change this design later on

File: private-templates-service/client/src/components/Dashboard/SearchPage/Filter/Filter.tsx (hunk 5 above)
Comment (nnmkhang) [reply private-templates-service/client/src/components/Dashboard/SearchPage/Filter/Filter.tsx]: We have not implemented any localization yet, we will be looking into it soon.

File: private-templates-service/client/src/components/Dashboard/SearchPage/Filter/Filter.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+const mapStateToProps = (state: RootState) => {
+  return {
+    isSearch: state.search.isSearch,
+    filterType: state.filter.filterType
+  }
+}
+
+const mapDispatchToProps = (dispatch: any) => {
+  return { 
+    clearFilter: () => {
```
[the remark this answers is not in the document]
Comment (nnmkhang) [reply private-templates-service/client/src/components/Dashboard/SearchPage/Filter/Filter.tsx]: I was just adding them as placeholders for when I implement the clearFilter/ clearSort

**Pull request merged**